onnx/defs/tensor/utils.cc

```diff
@@ -157,54 +165,53 @@ void resizeShapeInferenceVersioned(InferenceContext& ctx, int opset_version) {
   if (axes_attr) {
     axes = RetrieveValues<int64_t>(*axes_attr);
   }
+  if (hasSizesInput) {
+    if (!axes.empty()) {
+      if (sizes_data.size() != axes.size()) {
+        fail_shape_inference(
+            "Number of elements of input 'sizes' (",
+            sizes_data.size(),
+            ") does not match the number of axes (",
+            axes.size(),
+            ").");
+      }
 
-  if (nullptr != sizes) {
-    if (sizes->data_type() == TensorProto::INT64) {
-      auto sizes_data = ParseData<int64_t>(sizes);
-      if (!axes.empty()) {
-        if (sizes_data.size() != axes.size()) {
-          fail_shape_inference(
-              "Number of elements of input 'sizes' (",
-              sizes_data.size(),
-              ") does not match the number of axes (",
-              axes.size(),
-              ").");
-        }
-
-        std::vector<bool> tmp(rank_x, false);
-        for (auto axis : axes) {
-          if (tmp[axis]) {
-            fail_shape_inference("Repeated axis: ", axis);
-          }
-          tmp[axis] = true;
-        }
-      } else {
-        // sizes_data contains scales for all axes
-        if (sizes_data.size() != rank_x) {
-          fail_shape_inference("Number of elements of input 'sizes' must be same as rank of input 'X'");
+      std::vector<bool> tmp(rank_x, false);
+      for (auto axis : axes) {
+        if (tmp[axis]) {
+          fail_shape_inference("Repeated axis: ", axis);
         }
+        tmp[axis] = true;
+      }
+    } else {
+      // sizes_data contains scales for all axes
+      if (sizes_data.size() != rank_x) {
+        fail_shape_inference(
+            "Number of elements of input 'sizes' (",
+            sizes_data.size(),
+            ") must be same as rank of input 'X' (",
+            rank_x,
+            ").");
```

Comment:
Actually I was confused about this check. I saw the document also mentioned this, but existing test models like `test_resize_tf_crop_and_resize_axes_2_3` and `test_resize_tf_crop_and_resize_axes_3_2` do not follow this. Take `test_resize_tf_crop_and_resize_axes_2_3` as an example, the model is (input rank 4 != sizes 2):

![image](https://user-images.githubusercontent.com/14194980/195221692-ae4b451d-f84d-47df-a7db-d4183624bc67.png)

@gramalingam do you think they are invalid ONNX models?

Reply:
Hi, I think this may be a bug introduced when the `axes` attribute was added to Resize. If the `axes` attribute is specified, we need to account for it (and the PR that added the attribute may have failed to account for it both in code and the documentation).

Reply:
This PR: https://github.com/onnx/onnx/pull/4126

Reply:
Good catch!

Reply:
Thank you for the helpful context! Sorry that I think I provided the wrong model above. Here is the right model for `test_resize_tf_crop_and_resize_axes_2_3`.

![image](https://user-images.githubusercontent.com/14194980/195420523-9898babf-69e6-4790-b04e-bdae26b1f3d3.png)


Based on the naming of this test model, does it mean that this model should have axis attribute?

Reply:
So, it is difficult to see what was in the old-version vs. new-version here ... the check is fine if axes attribute wasn't specified, right? The new-version seems to have this check only if axes attribute is not specified?

Reply:
IIUC, the only difference about this issue is the update from this PR enables "sizes" input to have the shape information even if the input tensor is unknown. Previously the tensor (sizes) will be nullptr anyway so it did not check the rank for input x and input sizes... IMO, a possible fix might be making these 2 models have attribute axes (2, 3) and axes (3, 2) and then they should pass the existing check. Does it look good to you?

Reply:
I agree the test-cases seem to be missing the axes attribute. We should add them. I guess you are saying that the checker wasn't catching the error previously, and we are able to catch it now (because of the improvement due to data-propagation)?

Reply:
> I guess you are saying that the checker wasn't catching the error previously, and we are able to catch it now (because of the improvement due to data-propagation)?

Yes. data-propagation will at least keep the shape information (the shape of "sizes" in this case) so it can do more check. I will update these models then. Thanks!

Reply:
I have updated these 2 models with attribute axes. Now both of them have passed onnx.shape_inference. PTAL.